draft-js-inline-toolbar-plugin/src/components/Toolbar/index.js

```diff
@@ -45,25 +78,99 @@ export default class Toolbar extends React.Component {
    */
   onOverrideContent = (overrideContent) => {
     this.setState({ overrideContent });
-  }
+  };
 
   onSelectionChanged = () => {
     // need to wait a tick for window.getSelection() to be accurate
     // when focusing editor with already present selection
     setTimeout(() => {
       if (!this.toolbar) return;
-      const relativeParent = getRelativeParent(this.toolbar.parentElement);
-      const toolbarHeight = this.toolbar.clientHeight;
-      const relativeRect = (relativeParent || document.body).getBoundingClientRect();
+
+      if (typeof this.state.width !== 'number') {
+        this.setState({ width: this.toolbar.offsetWidth });
+      }
+
+      const metrics = {
+        rule: 'left',
+        ruleValue: 0,
+        shift: null,
+        shiftValue: 0
+      };
+      // boundings of the selected text
       const selectionRect = getVisibleSelectionRect(window);
+      const selection = this.props.store.getItem('getEditorState')().getSelection();
+      if (!selectionRect || selection.isCollapsed()) return;
 
-      if (!selectionRect) return;
+      const relativeParent = getRelativeParent(this.toolbar.parentElement);
+      const relativeRect = (relativeParent || document.body).getBoundingClientRect();
+      const windowWidth = document.documentElement.clientWidth;
+
+      // if parent block width is wider than the toolbar, we must forcibly set its
+      // height to prevent unexpected geometry changes
+      if (windowWidth > this.toolbar.offsetWidth) {
```

Comment:
I don't understand this problem. What do you mean by `parent block width`? If you're talking about draft js blocks then those are expected to be wider.

Reply:
Also, `height` is a typo here I guess?

Reply:
I'm agree, this is very odd comment. Honestly I do not remember why I wrote this :smile: and what this check should do. We just need to override `width` everytime.